dwave/optimization/src/nodes/_state.hpp

```diff
@@ -39,38 +39,39 @@ class ArrayStateData {
     explicit ArrayStateData(Range&& values) noexcept
             : ArrayStateData(std::vector<double>(values.begin(), values.end())) {}
 
-    // Assign new values to the state, tracking the changes from the previous state to the new
-    // one. Including resizes.
-    bool assign(std::ranges::sized_range auto&& values) {
-        // dev note: we could implement a version of this that doesn't need sized_range.
-        const ssize_t overlap_length = std::min<ssize_t>(buffer.size(), std::ranges::size(values));
+    // Assign new values to the state, starting from an offset and optionally "trimming" the
+    // end of the previous buffer if it extends past the range of the newly set values.
+    bool assign_partial(std::ranges::sized_range auto&& values, ssize_t offset,
+                        bool trim_end = false) {
```

Comment:
As far as I can tell, we never use `trim_end = False`. Is this legacy for an old implementation or?

Reply:
I had it in mind that this could be used by a few more nodes where we want to assign subrange of the buffer. But now I can't find any more examples so I can just remove the parameter for now.

Reply:
In that case I am tempted to keep `bool assign(..., ssize_t offset = 0)` rather than adding an overload. But could go either way